tests/test_cmdLineFwk.py

```diff
@@ -0,0 +1,216 @@
+    args = argparse.Namespace()
+    args.pipeline = pipeline
+    args.qgraph = qgraph
+    args.pipeline_actions = pipeline_actions
+    args.order_pipeline = order_pipeline
+    args.save_pipeline = save_pipeline
+    args.save_qgraph = save_qgraph
+    args.pipeline_dot = pipeline_dot
+    args.qgraph_dot = qgraph_dot
+    return args
+
+
+class CmdLineFwkTestCase(unittest.TestCase):
+    """A test case for CmdLineFwk
+    """
+
+    def testMakePipeline(self):
+        """Tests for CmdLineFwk.makePipeline method
+        """
+        fwk = CmdLineFwk()
+        taskFactory = TaskFactoryMock()
+
+        # make empty pipeline
+        args = _makeArgs()
+        pipeline = fwk.makePipeline(taskFactory, args)
+        self.assertIsInstance(pipeline, Pipeline)
+        self.assertEqual(len(pipeline), 0)
+
+        # few tests with pickle
+        with makeTmpFile() as tmpname:
+            # make empty pipeline and store it in a file
+            args = _makeArgs(save_pipeline=tmpname)
+            pipeline = fwk.makePipeline(taskFactory, args)
+            self.assertIsInstance(pipeline, Pipeline)
+
+            # read pipeline from a file
+            args = _makeArgs(pipeline=tmpname)
+            pipeline = fwk.makePipeline(taskFactory, args)
+            self.assertIsInstance(pipeline, Pipeline)
+            self.assertEqual(len(pipeline), 0)
+
+            # pickle with wrong object type
+            with open(tmpname, "wb") as pickleFile:
+                pickle.dump({}, pickleFile)
+            args = _makeArgs(pipeline=tmpname)
+            with self.assertRaises(TypeError):
+                pipeline = fwk.makePipeline(taskFactory, args)
+
+        # single task pipeline
+        actions = [
+            _PipelineAction(action="new_task", label="task1", value="TaskOne")
+        ]
+        args = _makeArgs(pipeline_actions=actions)
+        pipeline = fwk.makePipeline(taskFactory, args)
+        self.assertIsInstance(pipeline, Pipeline)
+        self.assertEqual(len(pipeline), 1)
+
+        # many task pipeline
+        actions = [
+            _PipelineAction(action="new_task", label="task1a", value="TaskOne"),
+            _PipelineAction(action="new_task", label="task2", value="TaskTwo"),
+            _PipelineAction(action="new_task", label="task1b", value="TaskOne")
+        ]
+        args = _makeArgs(pipeline_actions=actions)
+        pipeline = fwk.makePipeline(taskFactory, args)
+        self.assertIsInstance(pipeline, Pipeline)
+        self.assertEqual(len(pipeline), 3)
+
+    def testMakeGraphFromPickle(self):
+        """Tests for CmdLineFwk.makeGraph method.
+
+        Only most trivial case is tested that does not do actual graph
+        building.
+        """
+        fwk = CmdLineFwk()
+        taskFactory = TaskFactoryMock()
+
+        with makeTmpFile() as tmpname:
+
+            # make empty graph and store it in a file
+            qgraph = QuantumGraph()
+            with open(tmpname, "wb") as pickleFile:
```

Comment:
Would this be better to do like the test above where you using the save the graph by using the save_graph argument to _makeArgs?

Reply:
The `CmdLineFwk.makeGraph()` method cannot make empty QGraph, and to make non-empty one it needs butler which we do not have yet implemented for unit tests. So the only way to save QGraph to a pickle file now is to make an instance manually and write it ourselves. That is simple enough for this unit test.